### ISSUE TITLE
Support scss modules in new Web template

### ISSUE THREAD

**Issue by pavsaund** [dolittle-entropy/vanir#116]
There is an issue when building scss templates imported into ts / tsx files fie first time around (ex github build action), as tsc needs to generate the appropriate typings. This isn't the case for subsequent build. The workaround is to add a typings folder that declares scss modules. There may be a universal way of declaring these modules, from a shared Web folder, but I haven't been able to solve that.

Stack overflow question for reference: https://stackoverflow.com/questions/56563243/how-to-import-a-scss-file-into-a-typescript-file

**<ProjectName>/Web/typings/declarations.d.ts**
```ts
declare module '*.scss';
```

**Comment by einari** [comment]
After some research and testing, I found the following loader to be better than the one we have right now: 
https://www.npmjs.com/package/@teamsupercell/typings-for-css-modules-loader

At least this will get the builds working. What I can't really get to work is the initial generation when using devserver. 

There is one thing I find interesting though, there might be a way to configure it work the way we want.

**Comment by jakhog** [comment]
Maybe worth checking this out: https://github.com/mrmckeb/typescript-plugin-css-modules

It would be nice to not have the auto-generated files in there if we don't need to. As I understand it, we don't really need the declared modules unless we aim to publish things as packages. Webpack doesn't really care, so as long as we get some checking on the classes used it should be fine right?

**Comment by einari** [comment]
Yup.. Agree.  I don't like those d.ts files in the repo. And I'm also leaning towards that if possible, I'd rather just be able to import any .scss file and not having to explicitly use .module for those we want to import into TS - feels like a "leaky abstraction" :)

**Comment by pavsaund** [comment]
Using css modules should be the choice of the developer developing the feature, as pulling in css as modules isn't always the best solution. That said if I understand where you are coming from, I don't think you want to remove the decision from the developer, but instead you are talking about having a build system that pattern matches well enough to resolve if an import is for css-modules or not, and keep that completely transparent to the developer? They would then write scss code as usual and either import styles, or the entire scss file as needed. 

That said I don't think we NEED those d.ts files, but the do enable typesafety at build time, which to me is kinda valuable. It's not a deal breaker to lose that, and still have the scoping options of css-modules, but I prefer both. Especially since we have that right now :)

**Comment by einari** [comment]
Absolutely. The choice of importing them like this is completely up to the developer. Its just about lowering the threshold for doing so and creating a consistent dev-experience for it. Right now it feels a bit like its a second-class citizen since there are hoops to go through.
